plugins/huaweicloud/src/main/java/org/eclipse/xpanse/plugins/huaweicloud/price/HuaweiCloudGlobalPriceCalculator.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.plugins.huaweicloud.price;
+
+import com.huaweicloud.sdk.bssintl.v2.model.ListOnDemandResourceRatingsResponse;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.modules.models.billing.Price;
+import org.eclipse.xpanse.modules.models.billing.Resource;
+import org.eclipse.xpanse.modules.models.billing.enums.Currency;
+import org.eclipse.xpanse.modules.models.billing.enums.PricingPeriod;
+import org.eclipse.xpanse.modules.models.common.exceptions.ClientApiCallFailedException;
+import org.eclipse.xpanse.modules.orchestrator.price.ServiceFlavorPriceRequest;
+import org.eclipse.xpanse.plugins.huaweicloud.price.model.ProductInfo;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Class that implements the price calculation for the global rest apis of HuaweiCloud.
+ */
+@Slf4j
+@Component
+public class HuaweiCloudGlobalPriceCalculator {
+
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    @Value("${huaweicloud.price.calculator.global.restapi.url:https://portal-intl.huaweicloud.com"
```

Comment:
@baixinsui this is missing in the application.properties file?

Reply:
Submitted